src/library_webgl.js

```diff
@@ -3726,6 +3735,20 @@ var LibraryGL = {
   },
 #endif
 
+  _write_array: function(arr, dst, dstLength, heapType) {
```

Comment:
`writeArrayToMemory` serializes a given array to HEAPU8 only, i.e. as a sequence of bytes/ASCII characters. `_write_array` serializes a given array of floats or ints. If `writeArrayToMemory` was used here, those floats and ints would be getting truncated to 0-255 on assignment to HEAPU8.

Reply:
Ok, yes, the old function isn't good then.

If we think this is GL specific, how about `$writeGLArray`? Seems like this might be useful for other things in the GL area, so a more JS-friendly name feels better to me (C-style `_foo_bar` suggests something is really internal and should never be reused; if we disagree on that let's discuss more). Please also add a comment on the function explaining what it does (something like "write a JS typed array containing 32-bit floats or ints to memory" perhaps).